app/src/main/java/com/example/bigowlapp/viewModel/SignUpViewModel.java

```diff
@@ -93,4 +101,33 @@ public String randomNumberStringGenerator() {
         int randomNumber = (int) ((random.nextDouble() * (max - min + 1)) + min);
         return String.format(Locale.getDefault(), "%04d", randomNumber);
     }
+
+    public void verifySmsInvitationsCollection(String phoneNumber){
+        LiveDataWithStatus<List<SmsInvitationRequest>> phoneNumberSent = repositoryFacade.getSmsInvitationRepository().getListOfDocumentByAttribute("phoneNumberSent", phoneNumber, SmsInvitationRequest.class);
+        phoneNumberSent.observeForever(smsInvitationList -> {
+            if(smsInvitationList.isEmpty()){
+                return;
+            }
+
+            for (SmsInvitationRequest smsInvitationRequest : smsInvitationList) {
+                createNotificationObject(smsInvitationRequest.getSenderUid(), phoneNumber);
+            }
+        });
+    }
+
+    public void smsInvitationNotificationListener(Context context){
+        NotificationListenerManager invitationListener = new NotificationListenerManager();
+        invitationListener.listen(context);
+    }
+
+    public void createNotificationObject(String senderUid, String phoneNumber){
+
+        Notification newNotification = new SmsInvitationRequest();
```

Comment:
This should be a Notification instead of an SmsInvitationRequest.

Reply:
Done.